File: 백준/Silver/2670. 연속부분최대곱/연속부분최대곱.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
	
    public static void main(String[] args) throws IOException {
    	
    	BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    	
    	int N = Integer.parseInt(br.readLine());
    	
    	double mul = Double.parseDouble(br.readLine());
    	double max = mul;
    	for (int i=1; i<N; i++) {
    		double cur = Double.parseDouble(br.readLine());
    		mul = Math.max(mul*cur, cur);
    		max = Math.max(max, mul);
    	}
    	System.out.println(String.format("%.3f", max));
    	
    	
    }
    

        
}

File: 백준/Silver/2670. 연속부분최대곱/연속부분최대곱.py
N = int(input())

lst = [float(input()) for _ in range(N)]

mx = lst[0]
cur = mx
for f in lst[1:] :
    if f > cur*f :
        cur = f
    else :
        cur *= f
    mx = max(mx, cur)
print(f"{mx:.3f}")